src/api/cli/daphne.cpp

```diff
@@ -416,3 +430,37 @@ main(int argc, char** argv)
 
     return StatusCode::SUCCESS;
 }
+
+int startDistributedWorkers(int argc, char** argv){
+    MPIWorker worker;
+    worker.joinComputingTeam();
+    return StatusCode::SUCCESS;
+}
+
+
+int main(int argc, char** argv){
+    int id, size;
+    MPI_Init(NULL,NULL);
+    MPI_Comm_rank(MPI_COMM_WORLD, &id);
+    MPI_Comm_size(MPI_COMM_WORLD, &size);
+    char processor_name[MPI_MAX_PROCESSOR_NAME];
+    int name_len;
+    MPI_Get_processor_name(processor_name, &name_len);
+    std::cout<<"I am "<< id <<" out of "<< size << " reside on "<< processor_name <<std::endl;
+    int res;
+    if(id==COORDINATOR){
+        res=startCoordinator(argc, argv);
+        std::cout<<"==========Done=======\n";
+        unsigned char terminateMessage=0x00;
+        for(int i=1;i<size;i++){
+            MPI_Send(&terminateMessage,1, MPI_UNSIGNED_CHAR, i,  DETACH, MPI_COMM_WORLD);
+            std::cout<<"coordinator detached worker "<<i<<std::endl; 
+       }
+    }   
+    else{
+        res=startDistributedWorkers(argc, argv);
+    }   
+
+    MPI_Finalize();
+    return res;
+}
```

Comment:
All that MPI stuff should be optional imho. It should only be called upon request (config parameter or maybe --mpi or --api MPI ?)